renku/core/utils/templates.py

```diff
@@ -0,0 +1,871 @@
+    )
+
+    update_available = is_template_update_available(
+        template_source=template_source, template_version=template_version, project=project
+    )
+
+    return update_available, project.template_version, template_version
+
+
+def is_template_update_available(template_source, template_version, project) -> bool:
+    """Return True if template is newer than the project's template."""
+    if not project.template_version:
+        return False
+
+    if is_renku_template(template_source):
+        template_version = Version(template_version)
+        try:
+            current_version = Version(project.template_version)
+        except ValueError:  # NOTE: project.template_version is not a valid SemVer
+            return True
+        else:
+            return current_version < template_version
+    else:
+        # NOTE: Project's template commit cannot be newer than the remote template's HEAD, so, this check is enough to
+        # see if an update is available
+        return template_version != project.template_version
+
+
+def create_template_sentence(templates, describe=False, instructions=False):
+    """Create templates choice sentence.
+
+    :ref templates: list of templates coming from manifest file
+    :ref instructions: add instructions
+    """
+    RenkuTemplate = namedtuple("RenkuTemplate", ["index", "id", "description", "variables"])
+
+    def extract_description(template_elem):
+        """Extract description from template manifest."""
+        if describe:
+            return template_elem["description"]
+        return None
+
+    def extract_variables(template_elem):
+        """Extract variables from template manifest."""
+        if describe:
+            descriptions = []
+            for name, variable in template_elem.get("variables", {}).items():
+                variable_type = f', type: {variable["type"]}' if "type" in variable else ""
+                enum_values = f', options: {variable["enum"]}' if "enum" in variable else ""
+                default_value = f', default: {variable["default_value"]}' if "default_value" in variable else ""
+                description = variable["description"]
+
+                descriptions.append(f"{name}: {description}{variable_type}{enum_values}{default_value}")
+            return "\n".join(descriptions)
+
+        return ",".join(template_elem.get("variables", {}).keys())
+
+    templates_friendly = [
+        RenkuTemplate(
+            index=index + 1,
+            id=template_elem["folder"],
+            description=extract_description(template_elem),
+            variables=extract_variables(template_elem),
+        )
+        for index, template_elem in enumerate(templates)
+    ]
+
+    table_headers = OrderedDict((("index", "Index"), ("id", "Id"), ("variables", "Parameters")))
+
+    if describe:
+        table_headers["description"] = "Description"
+
+    text = tabulate(templates_friendly, headers=table_headers)
```

Comment:
It's used for showing template selection prompt. We can't move this part to CLI (without complicating code) since we don't know the list of available templates before we fetch them.